SmartDeviceLink/SDLIAPSession.m

```diff
@@ -141,10 +258,26 @@ - (SDLStreamErrorHandler)streamErroredHandler {
     };
 }
 
+- (SDLStreamHasSpaceHandler)streamHasSpaceHandler {
+    __weak typeof(self) weakSelf = self;
+    
+    return ^(NSStream *stream) {
+        __strong typeof(weakSelf) strongSelf = weakSelf;
+        
+        if (strongSelf.isDataSession) {
+            NSError *sendErr = nil;
+            
+            if (![strongSelf sdl_dequeueAndWriteToOutputStream:&sendErr] && sendErr != nil) {
+                [strongSelf sdl_handleOutputStreamWriteError:sendErr];
+            }
+        }
+    };
+}
 
 #pragma mark - Lifecycle Destruction
 
 - (void)dealloc {
+    self.sendDataQueue = nil;
```

Comment:
Has this been verified in develop with zombies on Instruments run? Many of the connection transition issues we have seen are due to the External Accessory Framework objects holding zombie references to partially deallocated objects. I am all for removing unnecessary code that ARC can generate itself, but it appears in the complex chain of events between the XPC communication from the app to the IAP daemon that object references are not cleaned up during transitions, leading to zombies when a new connection to the head unit is established while the old one is partially torn down.